agent/exec/container/executor.go

```diff
@@ -38,9 +38,25 @@ func (e *executor) Describe(ctx context.Context) (*api.NodeDescription, error) {
 	}
 
 	addPlugins("Volume", info.Plugins.Volume)
+
 	// Add builtin driver "overlay" (the only builtin multi-host driver) to
-	// the plugin list by default.
-	addPlugins("Network", append([]string{"overlay"}, info.Plugins.Network...))
+	// the plugin list if there is no "overlay" locally on that engine.
+
+	networks := append([]string{"overlay"}, info.Plugins.Network...)
+
+	// remove duplicated one in networks
+	network_map := map[string]struct{}{}
```

Comment:
Just make `addPlugins` do this. `api.PluginDescription` can be used as a map key:

```go
plugins := map[api.PluginDescription]struct{}{}
addPlugins := func(typ string, names []string) {
  for _, name := range names {
    plugins[api.PluginDescription{Type: type, Name: name}] = struct{}{}
  }
}
```

Then, before setting the field, unroll the map keys into a slice:

```go
pluginField := make([]api.PluginDescription], 0, len(plugins))
for k := range plugins {
  pluginField = append(pluginField, k)
}

// these should probably be sorted, as well, to keep output stable
sortPlugins(pluginField)
```